File: C++/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max = *max_element(candies.begin(), candies.end());
            
        vector<bool> ans;
        
        for (int c: candies) {
            ans.push_back(c + extraCandies >= max);
        }
        
        return ans;
    }
};

File: Kotlin/kids-with-the-greatest-number-of-candies.kt
class Solution {
    fun kidsWithCandies(candies: IntArray, extraCandies: Int): List<Boolean> {
        val maxCandies = candies.max()!!
        val ans = mutableListOf<Boolean>()
        for (c in candies) {
            ans.add((c + extraCandies) >= maxCandies)
        }
        return ans
    }
}